docs/assembly.rst

```diff
@@ -609,12 +609,21 @@ first.
 
 Solidity manages memory in a very simple way: There is a "free memory pointer"
 at position ``0x40`` in memory. If you want to allocate memory, just use the memory
-from that point on and update the pointer accordingly.
+starting from where this pointer points at and update it accordingly.
+There is no built-in mechanism to release or free allocated memory.
+Here is an assembly snippet that can be used for allocating memory::
+
+    function allocate(length) -> pos {
+      pos := mload(0x40)
+      mstore(0x40, add(pos, length))
+    }
 
 The first 64 bytes of memory can be used as "scratch space" for short-term
```

Comment:
If we want to be precise we would add a note here that this scratch space starts at `0x00`.

Likely this entire section should be rewritten in a better form. It shows that things have been added to it in iterations.

Reply:
Created an issue for it: #4728